src/Fantomas.Core/CodePrinter.fs

```diff
@@ -2172,20 +2172,25 @@ let (|EndsWithSingleListApp|_|) (config: FormatConfig) (appNode: ExprAppNode) =
         visit appNode.Arguments
 
 let (|EndsWithSingleRecordApp|_|) (config: FormatConfig) (appNode: ExprAppNode) =
-    if not config.ExperimentalStroustrupStyle then
-        None
-    else
-        let mutable otherArgs = ListCollector<Expr>()
+    let mutable otherArgs = ListCollector<Expr>()
 
-        let rec visit (args: Expr list) =
-            match args with
-            | [] -> None
-            | [ Expr.Record _ | Expr.AnonRecord _ as singleList ] -> Some(otherArgs.Close(), singleList)
-            | arg :: args ->
-                otherArgs.Add(arg)
-                visit args
+    let rec visit (args: Expr list) =
+        match args with
+        | [] -> None
+        | [ Expr.Record _ | Expr.AnonRecord _ as singleRecord ] -> Some(otherArgs.Close(), singleRecord)
+        | arg :: args ->
+            otherArgs.Add(arg)
+            visit args
 
+    if config.ExperimentalStroustrupStyle then
         visit appNode.Arguments
+    else
+        match appNode.FunctionExpr with
+        | Expr.Constant _ -> None
+        | ParenExpr _ -> None
+        | UppercaseExpr when config.StroustrupForMultilineRecordAsUppercaseInvocationFinalArg -> visit appNode.Arguments
+        | LowercaseExpr when config.StroustrupForMultilineRecordAsLowercaseInvocationFinalArg -> visit appNode.Arguments
```

Comment:
I'm still not convinced about bringing casing into this mix.
Does it really matter when the function name starts with a lower or capital?

Reply:
I guess not a ton, the distinction I was picturing was like if you wanted it to apply with Discriminated Union cases e.g. `Some`, where it's clear that it's not a computation expression, but didn't want to use it if it's lowercase to avoid ambiguity between application and computation expressions.

Reply:
The casing won't always tell you that much. 
Consider

```fsharp
open type System.Console

Write {|
    X = 2
|}

let printfObj = printfn "%A"

printfObj {|
    X = 2
|}

type VoidBuilder<'T>() =
    member x.Run (_ : 'T) = ()
    member x.Zero () = ()
    
let VOID = VoidBuilder()

VOID {
    ()
}
```

You cannot truly say what `appNode.FunctionExpr` will be, and if you want to use this style, I think you should go all in.

Perhaps this should not even be its own setting. For `EndsWithDualListApp`, `EndsWithSingleListApp`, I see a use-case (HTML DSL code) to use this outside of Stroustrup.

For `EndsWithSingleRecordApp`, it might just be a part of Stroustrup.

Would it be ok for you to discuss this further during our office hours?

Reply:
Fair points. Honestly I'm fine keeping it one setting. I may have been trying prematurely to optimize against a nonexistant argument haha.